misc/makefile_help.go

```diff
@@ -0,0 +1,381 @@
+// Package main implements a CLI tool to parse Makefile targets and scan directories.
+// It extracts targets (lines beginning with a letter and containing a colon),
+// captures inline comments (starting with one or more '#'), and optionally
+// expands '%' targets with wildcard values. It also scans provided directories
+// for Makefiles, flags those with a "help" target, and reads the first line of
+// any README.md there as a banner.
+//
+// Usage:
+//
+//	go run main.go [OPTIONS] <Makefile>
+//
+// Options:
+//
+//	-r, --relative-to PATH    Treat PATH as the relative invocation path
+//	-d, --dirs DIR [...]      List of directories to scan for Makefiles
+//	-w, --wildcard WILD [...] List of wildcard substitutions for '%' targets
+//	-h, --help                Show usage and exit
+package main
+
+import (
+	"bufio"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+var (
+	numPrefixRE        = regexp.MustCompile(`^(\d+):`)
+	percentNumPrefixRE = regexp.MustCompile(`^%(\d+):`)
+)
+
+// flagList accumulates multiple flag values into a slice.
+type flagList []string
+
+// String returns the comma‑separated flag values.
+func (f *flagList) String() string {
+	var b strings.Builder
+	b.WriteByte('[')
+
+	for i, s := range *f {
+		if i > 0 {
+			b.WriteString(", ")
+		}
+		b.WriteString(strconv.Quote(s))
+	}
+
+	b.WriteByte(']')
+	return b.String()
+}
+
+// Set appends a new value to the flag list.
+func (f *flagList) Set(val string) error {
+	*f = append(*f, val)
+	return nil
+}
+
+// flagList accumulates multiple flag values into a slice.
+type nestedFlagList [][]string
```

Comment:
as said previously, i think we can go for a simple "search-and-replace" mechanism, so this type shouldn't be needed